### PR TITLE
Fix navigation editor

### LINKED ISSUE #28076: The navigation editor has encountered an unexpected error.
**Describe the bug**
When enabling and opening the (experimental) Gutenberg Navigation editor, 
it doesn't load with an error popup being shown: `The navigation editor has encountered an unexpected error.`
'Attempting recovery' using the provided button doesn't change anything.
The JavaScript error console logs an error that could be the underlying cause:
````
react_devtools_backend.js:2430 TypeError: Cannot read property 'icon' of null
    at e (index.js?ver=97fffcbb2ca23799ffea84ef0015b694:58)
    at we (react-dom.min.de439aae.js:84)
    at nh (react-dom.min.de439aae.js:96)
    at zj (react-dom.min.de439aae.js:230)
    at Th (react-dom.min.de439aae.js:152)
    at tj (react-dom.min.de439aae.js:152)
    at Te (react-dom.min.de439aae.js:146)
    at react-dom.min.de439aae.js:61
    at unstable_runWithPriority (react.min.e713ea3b.js:25)
    at Da (react-dom.min.de439aae.js:60)
````

**To reproduce**
1. Install and enable latests Gutenberg plugin; go to the Gutenberg experiments page (Admin area; item "Gutenberg" in sidebar; sub-item "Experiments"), check "Navigation view" and save the settings.
2. Go to the Gutenberg Navigation view admin page (Admin area; "Gutenberg" in sidebar; sub-item "Navigation (Beta)").
The navigation view seems to initialize, but then it doesn't load and an error popup is displayed: `The navigation editor has encountered an unexpected error.`.
(3.) Select the `Attempt recovery` button, notice that nothing changes.

**Expected behavior**
The Gutenberg Navigation view should load and be usable.

**Editor version (please complete the following information):**
- WordPress `5.6`. 
- Gutenberg plugin `9.7.0`.

**Desktop (please complete the following information):**
 - OS: Windows 10 x64
 - Browser: Chrome (`87.0.4280.141 (official build) (64-Bit)`)

### PR DESCRIPTION
**Pull request by talldan**
## Description
Fixes #28076.

The experimental navigation screen currently errors when trying to open it.

The `blockInformation` isn't defined straight away, but that's not handled gracefully. I imagine this wasn't noticed because in other editors ListView only renders when the List View dropdown is opened, compared to the navigation screen where ListView is always rendered and visible.

## How has this been tested?
Added an e2e test.

Manual testing
1. Enable the Navigation Screen experiment
2. Browse to the navigation screen, it should open
3. Try creating a menu

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-navigation/src/components/layout/index.js [hunk 1]
```diff
@@ -66,6 +66,7 @@ export default function Layout( { blockEditorSettings } ) {
 								templateLock: 'all',
 								hasFixedToolbar: true,
 							} }
+							useSubRegistry={ false }
```
Comment (talldan) [inline packages/edit-navigation/src/components/layout/index.js:69]: This is needed, as without this `wp.data.select( 'core/block-editor' ).getBlocks()` isn't usable—but it's needed in the test to assert which blocks are present.

Comment (draganescu) [reply packages/edit-navigation/src/components/layout/index.js:69]: Just wanted to whine here. These true / false flags are so hard to pin down on meaning when reading code.  Looking [here](https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/provider/README.md) it's not even mentioned.

**Review by draganescu — APPROVED** [review]
Tested and the change makes the screen work as expected. The code and added test are 🆗

**Comment by MarvelMoe** [comment]
Hello @draganescu . Im getting this navigation screen error with the latest version of Gutenberg installed and the Block-Based Bosco theme and Attempt Recovery is not fixing it. Im attaching a screenshot of the error, can you please advise on the exact directory path for where I can update the code to fix this.   

Screen Shot 2021-01-19 at 1 48 42 PM

**Comment by talldan** [comment]
@MarvelMoe Thanks for reaching out. The fix for this will be released in Gutenberg 9.8.

Fixing manually is a little tricky as the code is compiled when shipped.

**Pull request merged**